### PR TITLE
docs: Website documentation for vllm inferencing using rayserve on AWS Inferentia

### PR DESCRIPTION
**Pull request by sindhupalakodety**
…n AWS Inferentia

### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

 Created a website documentation for vllm inferencing using rayserve on AWS Inferentia

### Motivation

To have users use vllm in association with RayServe and Trainium and to use llmperf tool for benchmarking

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md [hunk 1]
```diff
@@ -0,0 +1,565 @@
+---
+title: RayServe with vLLM
```
Comment (askulkarni2) [inline website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md:2]: Let's keep the same convention as others focussing on model and accelerator.

```suggestion
title: Llama3 on Inferentia2
```

File: website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+
+### What is AWS Neuron?
+
+In this tutorial, you'll leverage [AWS Neuron](https://aws.amazon.com/machine-learning/neuron/), a powerful SDK that optimizes deep learning performance on AWS Inferentia and Trainium accelerators. Neuron seamlessly integrates with frameworks like PyTorch and TensorFlow, providing a comprehensive toolkit for developing, profiling, and deploying high-performance machine learning models on specialized EC2 instances such as Inf1, Inf2, Trn1, and Trn1n.
+
+### What is vLLM?
+
+[vLLM](https://docs.vllm.ai/en/latest/) is a high-performance library for LLM inference and serving, designed to maximize throughput and minimize latency. At its core, vLLM utilizes PagedAttention, an innovative attention algorithm that dramatically improves memory efficiency, allowing for optimal utilization of GPU resources. This open-source solution offers seamless integration through its Python API and OpenAI-compatible server, enabling developers to deploy and scale large language models like Llama 3 with unprecedented efficiency in production environments.
```
Comment (askulkarni2) [inline website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md:36]:
```suggestion
[vLLM](https://docs.vllm.ai/en/latest/) is a high-performance library for LLM inference and serving, designed to maximize throughput and minimize latency. At its core, vLLM utilizes [PagedAttention](https://docs.vllm.ai/en/latest/dev/kernel/paged_attention.html), an innovative attention algorithm that dramatically improves memory efficiency, allowing for optimal utilization of GPU resources. This open-source solution offers seamless integration through its Python API and OpenAI-compatible server, enabling developers to deploy and scale large language models like Llama 3 with unprecedented efficiency in production environments.
```

**Review by askulkarni2 — COMMENTED** [review]
@sindhupalakodety thanks for the PR! Minor comments. Also for next time, please open an issue in the repo for tracking.

**Review by vara-bonthu — COMMENTED** [review]
Thanks for the doc PR 👍🏼 Left few comments to be addressed.

File: website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+### Verify Neuron Plugin
+
+Neuron device plugin exposes Neuron cores & devices to kubernetes as a resource. Verify the status of the plugin installed by the blueprint. 
+
+```bash
+kubectl get ds neuron-device-plugin-daemonset --namespace kube-system
+```
+```bash
+NAME                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
+neuron-device-plugin-daemonset   2         2         2       2            2           <none>          2d2h
```
Comment (ratnopamc) [inline website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md:138]: Please change the daemonset name from `neuron-device-plugin-daemonset` to `neuron-device-plugin`.

File: website/docs/gen-ai/inference/img/ray_on_kubernetes.png
Comment (ratnopamc) [inline website/docs/gen-ai/inference/img/ray_on_kubernetes.png:1]: Please remove this image.

File: website/docs/gen-ai/inference/Neuron/llama3-inf2.md
[the remark this answers is not in the document]
Comment (sindhupalakodety) [reply website/docs/gen-ai/inference/Neuron/llama3-inf2.md:1]: Hi Vara, this is the file I changed. Changed the name of it to RayServe with vLLM .. do you want a copy of the original file to be in the repo.

File: website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md (hunk 1 above)
Comment (sindhupalakodety) [reply website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md:2]: Hi Apoorva, Vara recommended me to change the title to "RayServe with vLLM".

File: website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md (hunk 3 above)
Comment (sindhupalakodety) [reply website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md:138]: Sure will work on it.

File: website/docs/gen-ai/inference/img/ray_on_kubernetes.png
Comment (sindhupalakodety) [reply website/docs/gen-ai/inference/img/ray_on_kubernetes.png:1]: Sure. Will work on it.

File: website/docs/gen-ai/inference/img/llama3-inf2-architecture-vllm.png
[the remark this answers is not in the document]
Comment (sindhupalakodety) [reply website/docs/gen-ai/inference/img/llama3-inf2-architecture-vllm.png:1]: Sure. Will work on it.

File: website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md (hunk 1 above)
Comment (vara-bonthu) [reply website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md]: I have updated and aligned the names properly

**Review by ratnopamc — APPROVED** [review]
lgtm.

**Pull request merged**